mesonbuild/envconfig.py

```diff
@@ -355,6 +356,17 @@ def is_irix(self) -> bool:
         """Machine is IRIX?"""
         return self.system.startswith('irix')
 
+    def is_qnx(self) -> bool:
+        """Machine is QNX?"""
+        return self.system == 'qnx'
+
+    def is_aix(self) -> bool:
+        """Machine is AIX?"""
+        return self.system == 'aix'
+
+    def is_wsl(self) -> bool:
+        return self.is_linux() and 'microsoft' in platform.release().lower()
```

Comment:
My thought was that I'd really like to not have the `mesonlib.is_*` methods at all, because they often are reached for when they're incorrect because they're easy. In pretty much every case we use them except in our own test framework they're either wrong, or we have access to `environment.machines` anyway, but there's a couple of big changes required to make that possible after all of this work (the linkers and compilers need to have an `Environment` reference internally.)